src/runtime/aarch64_cpu_features.cpp

```diff
@@ -107,30 +107,33 @@ void set_platform_features(CpuFeatures &features) {
 
 namespace {
 
-void set_platform_features(CpuFeatures &) {
+void set_platform_features(CpuFeatures *) {
 }
 
 }  // namespace
 
 #endif
 
-WEAK CpuFeatures halide_get_cpu_features() {
-    CpuFeatures features;
-    features.set_known(halide_target_feature_arm_dot_prod);
-    features.set_known(halide_target_feature_arm_fp16);
-    features.set_known(halide_target_feature_armv7s);
-    features.set_known(halide_target_feature_no_neon);
-    features.set_known(halide_target_feature_sve);
-    features.set_known(halide_target_feature_sve2);
+}  // namespace Internal
+}  // namespace Runtime
+}  // namespace Halide
+
+extern "C" {
+
+WEAK int halide_get_cpu_features(Halide::Runtime::Internal::CpuFeatures *features) {
```

Comment:
Might be better to keep the function in Halide::Runtime::Internal to avoid all this verbosity. If it's extern "C" the mangling will be the same either way (i.e. the namespace doesn't affect the mangling).

Reply:
Okay! I’ll move it back into the namespace.